src/main/java/dot/tasks/TaskList.java

```diff
@@ -37,39 +38,12 @@ protected TaskList(int maxSize, Storage storage) {
     }
 
     protected TaskList(int maxSize, ArrayList<Task> tasks, Storage storage) {
+        // We will set maxSize to 2x of length of tasks, or maxSize, whichever is larger
         this.tasks = tasks;
-        this.maxSize = maxSize;
+        this.maxSize = Math.max(tasks.size() * 2, maxSize);
```

Comment:
using this is not necessary as it does not lead to confusion

Reply:
Thanks for the review! The fields here are being shadowed by constructor parameters.